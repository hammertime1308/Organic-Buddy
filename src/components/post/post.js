import React from 'react';
import { Text, Image } from 'react-native';
import Moment from 'moment';

import {
  StyledPost,
  RowContainer,
  ColumnContainer,
  Title,
  Description,
  MetaDataContainer,
  MetaDataText,
} from './styles';

export const Post = ({
  postId = '',
  userId = '',
  title = '',
  description = '',
  image = [],
  timestamp = '',
  comments = ['comment'],
}) => {
  Moment.locale('en');
  let date = Moment(timestamp).format('MMM Do YY');
  return (
    <StyledPost activeOpacity={0.9} onPress={() => alert('Post pressed')}>
      <RowContainer>
        {image.length !== 0 ? (
          <Image
            style={{
              flex: 1,
              height: 120,
              width: '100%',
              resizeMode: 'stretch',
              marginRight: 10,
            }}
            source={{ uri: image[0] }}
          />
        ) : (
          <Text />
        )}
        <ColumnContainer>
          <Title>{title}</Title>
          <Description>{description}</Description>
          <MetaDataContainer>
            <MetaDataText>{date}</MetaDataText>
            <MetaDataText>Replies: {comments.length - 1}</MetaDataText>
          </MetaDataContainer>
        </ColumnContainer>
      </RowContainer>
    </StyledPost>
  );
};
