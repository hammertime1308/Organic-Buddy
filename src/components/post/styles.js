import styled from 'styled-components/native';

import { COLOR, FONT } from '../../styles';

export const StyledPost = styled.TouchableOpacity`
  margin: 10px 10px 0px 10px;
  border-radius: 5px;

  backgroundColor: ${COLOR.White};
  shadowColor: ${COLOR.Black};
  shadowOpacity: 0.26;
  shadowOffset: { width: 0, height: 2};
  elevation: 3;
  overflow: hidden;
`;

export const RowContainer = styled.View`
  display: flex;
  flex-direction: row;
  margin: 15px;
`;

export const ColumnContainer = styled.View`
  flex: 2;
  flex-direction: column;
`;

export const Title = styled.Text`
  flex: 2;
  font-size: ${FONT.FontSizeHeader};
  font-weight: ${FONT.FontWeightBold};
`;

export const Description = styled.Text`
  flex: 1;
  font-size: ${FONT.FontSizeNormal};
  font-weight: ${FONT.FontWeightSlightBold};
`;

export const MetaDataContainer = styled.View`
  flex: 1;
  flex-direction: row;
  align-items: flex-end;
`;

export const MetaDataText = styled.Text`
  margin-right: 5px;
  font-size: ${FONT.FontSizeSmall};
  font-weight: ${FONT.FontWeightSlightBold};
  color: ${COLOR.GrayColor};
`;
