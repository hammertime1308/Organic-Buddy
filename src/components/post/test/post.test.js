import React from 'react';
import renderer from 'react-test-renderer';
import { Post } from '../post';

describe('Post', () => {
  it('Should render post without image', () => {
    const tree = renderer
      .create(
        <Post
          postId={'1'}
          userId={'111'}
          title={'Sample title for post'}
          description={'sample description for post'}
          image={[]}
          timestamp={'2021-06-12T10:17:52Z'}
          comments={['comment']}
          setCount={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render post with image', () => {
    const tree = renderer
      .create(
        <Post
          postId={'1'}
          userId={'111'}
          title={'Sample title for post'}
          description={'sample description for post'}
          image={[
            'https://cdn.pixabay.com/photo/2017/07/19/09/51/hallstatt-2518544_960_720.jpg',
          ]}
          timestamp={'2021-06-12T10:17:52Z'}
          comments={['comment']}
          setCount={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render post with multiple comments', () => {
    const tree = renderer
      .create(
        <Post
          postId={'1'}
          userId={'111'}
          title={'Sample title for post'}
          description={'sample description for post'}
          image={[]}
          timestamp={'2021-06-12T10:17:52Z'}
          comments={['comment', 'sample comment 1', 'sample comment 2']}
          setCount={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
