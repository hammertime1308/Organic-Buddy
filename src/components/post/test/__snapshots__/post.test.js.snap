// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Should render post with image 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffff",
      "borderRadius": 5,
      "elevation": 3,
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOpacity": 0.26,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 15,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn.pixabay.com/photo/2017/07/19/09/51/hallstatt-2518544_960_720.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 120,
          "marginRight": 10,
          "resizeMode": "stretch",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 2,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 2,
              "flexShrink": 1,
              "fontSize": 20,
              "fontWeight": "700",
            },
          ]
        }
      >
        Sample title for post
      </Text>
      <Text
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 15,
              "fontWeight": "600",
            },
          ]
        }
      >
        sample description for post
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgba(191, 191, 191, 1)",
                "fontSize": 12,
                "fontWeight": "600",
                "marginRight": 5,
              },
            ]
          }
        >
          2021-06-12 03:47:52
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(191, 191, 191, 1)",
                "fontSize": 12,
                "fontWeight": "600",
                "marginRight": 5,
              },
            ]
          }
        >
          Replies: 
          0
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Post Should render post with multiple comments 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffff",
      "borderRadius": 5,
      "elevation": 3,
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOpacity": 0.26,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 15,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "height": 120,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 2,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 2,
              "flexShrink": 1,
              "fontSize": 20,
              "fontWeight": "700",
            },
          ]
        }
      >
        Sample title for post
      </Text>
      <Text
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 15,
              "fontWeight": "600",
            },
          ]
        }
      >
        sample description for post
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgba(191, 191, 191, 1)",
                "fontSize": 12,
                "fontWeight": "600",
                "marginRight": 5,
              },
            ]
          }
        >
          2021-06-12 03:47:52
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(191, 191, 191, 1)",
                "fontSize": 12,
                "fontWeight": "600",
                "marginRight": 5,
              },
            ]
          }
        >
          Replies: 
          2
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Post Should render post without image 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffff",
      "borderRadius": 5,
      "elevation": 3,
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOpacity": 0.26,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 15,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "height": 120,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 2,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 2,
              "flexShrink": 1,
              "fontSize": 20,
              "fontWeight": "700",
            },
          ]
        }
      >
        Sample title for post
      </Text>
      <Text
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 15,
              "fontWeight": "600",
            },
          ]
        }
      >
        sample description for post
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgba(191, 191, 191, 1)",
                "fontSize": 12,
                "fontWeight": "600",
                "marginRight": 5,
              },
            ]
          }
        >
          2021-06-12 03:47:52
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(191, 191, 191, 1)",
                "fontSize": 12,
                "fontWeight": "600",
                "marginRight": 5,
              },
            ]
          }
        >
          Replies: 
          0
        </Text>
      </View>
    </View>
  </View>
</View>
`;
